### PR TITLE
Display columns on Record Board

### PR DESCRIPTION
**Pull request by charlesBochet**
## Context
We are refactoring the record board which has been hard coded in the codebase to display opportunities.

## What
In this PR:

- adding state select field type on opportunity standard object
- Introducing states to store columns: columnIds, columnFamilyState, columnFamilySelector (to ease usage), isColumnFirstFamilyState, isColumnLastFamilyState. We are introducing these familyState to guarantee good performance. A column should be able to read its columnInformation but do not re-render when other columns are changed.
- using this state to compute board columns based on select field options (the first we find for now)
- Re-adding columnHeader, columnMenu components but only redirecting to Settings to edit Select options

## Next steps:

- Load opportunities by column and display them through basic cards
- re-apply filters and sort
- re-apply viewFields
- pagination (performance)
- action bar, context menu, selection

### CODE REVIEW

File: packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexBoardContainerEffect.tsx [hunk 1]
```diff
@@ -1,11 +1,50 @@
...
+  });
+
+  const { objectMetadataItem } = useObjectMetadataItem({
+    objectNameSingular,
+  });
+
+  const navigate = useNavigate();
+
+  const navigateToSelectSettings = useCallback(() => {
+    navigate(`/settings/objects/${objectNamePlural}`);
```
Comment (thomtrp) [inline packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexBoardContainerEffect.tsx:30]: To store in `AppPath`?

File: packages/twenty-front/src/modules/object-record/record-board/hooks/internal/useSetRecordBoardColumns.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    useRecordBoardStates(recordBoardId);
+
+  const setRecordBoardColumns = useRecoilCallback(
+    ({ set, snapshot }) =>
+      (columns: RecordBoardColumnDefinition[]) => {
+        const currentColumns = snapshot
+          .getLoadable(getColumnIdsState())
+          .getValue();
+
+        const columnIds = columns.map(({ id }) => id);
```
Comment (Weiko) [inline packages/twenty-front/src/modules/object-record/record-board/hooks/internal/useSetRecordBoardColumns.ts:18]: I prefer this pattern but if it's not widely used you can ignore!
```suggestion
        const columnIds = columns.map(column => column.id)
```

**Review by thomtrp — APPROVED** [review]
LGTM, minor comments

**Review by Weiko — APPROVED** [review]
LGTM

**Review by Weiko — APPROVED** [review]

**Pull request merged**